src/modules/universal-discovery/components/bookmarks/bookmarks.component.js

```diff
@@ -0,0 +1,174 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import TableComponent from '../table/table.component';
+import { loadBookmarks } from '../../services/universal.discovery.service';
+
+import './css/bookmarks.component.css';
+
+export default class BookmarksComponent extends Component {
+    constructor() {
+        super();
+
+        this.state = {
+            items: [],
+            isLoading: false
+        };
+
+        this.updateItemsState = this.updateItemsState.bind(this);
+        this.bookmarksContent = this.bookmarksContent.bind(this);
+    }
+
+    componentDidMount() {
+        this.bookmarksContent();
+    }
+
+    /**
+     * Called when bookmark was removed
+     *
+     * @method onBookmarkRemoved
+     * @param {Object} itemLocation
+     * @memberof BookmarksComponent
+     */
+    onBookmarkRemoved(itemLocation) {
+        this.removeItem(itemLocation);
+    }
+
+    /**
+     * Removes item from list
+     *
+     * @method removeItem
+     * @param {Object} itemToRemoveLocation
+     * @memberof BookmarksComponent
+     */
+    removeItem(itemToRemoveLocation) {
+        const areTheSameItems = (fstItemLocation, sndItemLocation) => fstItemLocation.id === sndItemLocation.id;
+
+        this.setState(state => Object.assign({}, state, {
+            items: state.items.filter((item) => !areTheSameItems(item.value.Location, itemToRemoveLocation))
+        }));
+    }
+
+    /**
+     * Creates bookmarks loading error notification
+     *
+     * @method createBookmarksLoadingErrorNotification
+     * @memberof BookmarksComponent
+     */
+    createBookmarksLoadingErrorNotification() {
+        const bookmarksLoadingErrorEvent = new CustomEvent('ez-notify', {
+            detail: {
+                label: 'danger',
+                message: 'Bookmarks loading error.'
+            }
+        });
+
+        document.body.dispatchEvent(bookmarksLoadingErrorEvent);
+    }
+
+    /**
+     * Loads bookmarks
+     *
+     * @method bookmarksContent
+     * @memberof BookmarksComponent
+     */
+    bookmarksContent() {
+        this.setState(state => Object.assign({}, state, { isLoading: true }), () => {
+            const { restInfo } = this.props;
+            const bookmarksLoaded = new Promise(resolve => loadBookmarks(restInfo, resolve));
+
+            bookmarksLoaded
+                .then(this.updateItemsState)
+                .catch(this.createBookmarksLoadingErrorNotification);
+        });
+    }
+
+    /**
+     * Updates items state with bookmarks results
+     *
+     * @param {Object} response content query REST endpoint response
+     * @memberof BookmarksComponent
+     */
+    updateItemsState(response) {
+        this.setState(state => Object.assign({}, state, {
+            items: response.View.Result.searchHits.searchHit,
+            isLoading: false
+        }));
+    }
+
+    getTableLabels() {
```

Comment:
Missing doc

Reply:
Why don't you pass labels directly, but you're rebuilding its structure to something else?
If a structure is incorrect, then maybe you should correct entry structure if possible?